### PR TITLE
feat: Updated GraphQL form UI

### LINKED ISSUE #35494: [Task]: Add Zone & Section to GraphQL plugin form

### LINKED ISSUE #36410: [Bug]: REST API panel too many parameters not supported scroll
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I in Queries tab create new REST API and add too many parameters or headers beyond screen size, this panel not supported scroll, i can't add more parameters.

### Steps To Reproduce

1. Go to Page Queries
2. Add new Query, Select REST API
3. Add more headers exceeds panel size
4. Can't scroll to add more headers

### Public Sample App

https://app.appsmith.com/app/untitled-application-3/page1-66db00d0bd006624fdcc7449

### Environment

Production

### Severity

Low (Cosmetic UI issues)

### Issue video log

_No response_

### Version

Cloud Appsmith 3a8cbdfb

### LINKED ISSUE #36499: [Task]: Section componet to add property to take full width and height

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

This PR has following new features and fixes
1. Feat: Update GraphQL to use form control dynamic text field instead of custom implementation.
2. Feat: Added full width support for section component in action form.
3. Fix: Rest API headers not scrollable if there are more number of items.

EE PR: https://github.com/appsmithorg/appsmith-ee/pull/5297

Fixes #35494
Fixes #36410
Fixes #36499

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11240771681>
> Commit: 14b3e0020d312463eab177ace625a8ca705b4629
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11240771681&attempt=2" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Wed, 09 Oct 2024 05:42:42 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Release Notes

- **New Features**
	- Introduced a streamlined form-based design for the `PostBodyData` component, enhancing user interaction with dynamic text fields for GraphQL queries and variables.
	- Added a new styled component, `MainContainer`, to improve layout responsiveness in the `CommonEditorForm`.
	- Enhanced the `Section` component with an optional `isFullWidth` property for greater configurability.

- **Bug Fixes**
	- Adjusted the `LoadingContainer` height dynamically to enhance UI responsiveness.

- **Style**
	- Updated styles for the `Section` component to support full-width expansion when applicable.

- **Removed Features**
	- Deprecated the `QueryEditor` and `VariableEditor` components, simplifying the editor interface.

### CODE REVIEW

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/pages/Editor/APIEditor/Editor.tsx [hunk 1]
```diff
@@ -226,7 +227,7 @@ class ApiEditor extends React.Component<Props> {
 
 const formStyles: CSSProperties = {
   position: "relative",
-  height: "100%",
+  height: `calc(100% - ${EDITOR_TABS_HEIGHT})`,
```
[the remark this answers is not in the document]
Comment (hetunandu) [reply app/client/src/pages/Editor/APIEditor/Editor.tsx:230]: Screenshot 2024-10-08 at 9 10 14 AM

This is from modules

File: app/client/src/pages/Editor/ActionForm/Section/index.tsx [hunk 2]
```diff
@@ -18,6 +20,7 @@ const Section: React.FC<SectionProps> = ({
   return (
     <div
       className={classNames}
+      data-fullwidth={isFullWidth.toString()}
```
Comment (ashit-rath) [inline app/client/src/pages/Editor/ActionForm/Section/index.tsx:23]: Any reason for not using classes to control css states?

Comment (albinAppsmith) [reply app/client/src/pages/Editor/ActionForm/Section/index.tsx:23]: Good point... This was a practise we used in ADS as well I see in WDS. @KelvinOm Do you see any specific reason we should stick to this? As I can see, class is better performant than data attributes.

**Review by hetunandu — APPROVED** [review]

**Pull request merged**